src/scripts/slack/message_test.go

```diff
@@ -0,0 +1,110 @@
+package slack
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/assert/cmp"
+)
+
+// Check Slack API docs to come up with a list of responses we support and then add test here
+// ie Channel does not exist etc. 
+// validation for slack notification package
+func TestPostMessage(t *testing.T) {
+	// Create a new Message
+	s := &Message{
+		AccessToken:  "test_token",
+		Template:     "{}",
+		IgnoreErrors: true,
+	}
+
+	// Activate the httpmock
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
```

Comment:
rather than doing the reset in a bare `defer`, golang has a test cleanup feature that lets you register hook functions that will run after a test:
```suggestion
	t.Cleanup(func() {
		httpmock.DeactivateAndReset()
	})
```